compiler/rustc_const_eval/src/interpret/cast.rs

```diff
@@ -225,6 +231,22 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         }
     }
 
+    fn ptr_to_metadata(
+        &self,
+        src: &ImmTy<'tcx, M::Provenance>,
+        cast_to: TyAndLayout<'tcx>,
+    ) -> InterpResult<'tcx, ImmTy<'tcx, M::Provenance>> {
+        assert!(src.layout.ty.is_unsafe_ptr());
+        return Ok(match **src {
```

Comment:
This is almost a duplicate of `to_scalar_and_meta`. Seems like it would make sense to change `to_scalar_and_meta` to raise `InvalidUninitBytes` on uninit, and then we can directly use it here.